packages/contracts/contracts/escrow/CollateralEscrowV1.sol

```diff
@@ -0,0 +1,245 @@
+// SPDX-Licence-Identifier: MIT
+pragma solidity >=0.8.0 <0.9.0;
+
+// Contracts
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+
+// Interfaces
+import "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+import {
+    SafeERC20Upgradeable
+} from "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC721/IERC721Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC1155/IERC1155Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC721/IERC721ReceiverUpgradeable.sol";
+import "../interfaces/escrow/ICollateralEscrowV1.sol";
+
+contract CollateralEscrowV1 is OwnableUpgradeable, ICollateralEscrowV1 {
+    uint256 public bidId;
+    /* Mappings */
+    mapping(address => Collateral) public collateralBalances; // collateral address -> collateral
+
+    /* Events */
+    event CollateralDeposited(address _collateralAddress, uint256 _amount);
+    event CollateralWithdrawn(
+        address _collateralAddress,
+        uint256 _amount,
+        address _recipient
+    );
+
+    /**
+     * @notice Initializes an escrow.
+     * @notice The id of the associated bid.
+     */
+    function initialize(uint256 _bidId) public initializer {
+        require(owner() == address(0), "Escrow already initialized");
```

Comment:
can remove this require since the initializer modifer already takes care of this